src/core/thread/mesh_forwarder.cpp

```diff
@@ -1281,6 +1293,71 @@ otError MeshForwarder::GetFramePriority(const uint8_t *     aFrame,
     return error;
 }
 
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+otError MeshForwarder::SendEmptyMessage(void)
+{
+    otError  error   = OT_ERROR_NONE;
+    Message *message = nullptr;
+
+    VerifyOrExit(mEnabled && !Get<Mac::Mac>().GetRxOnWhenIdle() &&
+                     Get<Mle::MleRouter>().GetParent().IsStateValidOrRestoring(),
+                 error = OT_ERROR_INVALID_STATE);
+
+    message = Get<MessagePool>().New(Message::kTypeMacEmptyData, 0);
+    VerifyOrExit(message != nullptr, error = OT_ERROR_NO_BUFS);
+
+    SuccessOrExit(error = SendMessage(*message));
+
+exit:
+    if (error != OT_ERROR_NONE && message != nullptr)
+    {
+        message->Free();
+    }
+    otLogDebgMac("Send empty message, error:%d", error);
+    return error;
+}
+
+void MeshForwarder::PrepareEmptyDataFrame(Mac::TxFrame &aFrame)
```

Comment:
It is quite similar to existing `IndirectSender::PrepareEmptyFrame(Mac::TxFrame &aFrame, Child &aChild, bool aAckRequest)` , better to have a common function to avoid redundancy?

Reply:
Was about to suggest the same thing. I fully agree. Should re-use the method.